### ISSUE TITLE
GreyboxFuzzer error

### ISSUE THREAD

**Issue by nguyenthanhvuh** [uds-se/fuzzingbook#81]
When running the GreyboxFuzzer chapter (via Jupyter), I cannot get through the code
```
cg = get_callgraph(maze_code)
...
```

The full error is listed below.   Also, is there a "stable" version of the book ? I see that the current Master branch is updated frequently and it might break things,  which makes it very inconvenient for students who just want to see how things work.

The error is :  
```
---------------------------------------------------------------------------
ParseException                            Traceback (most recent call last)
/usr/local/lib/python3.8/site-packages/dot_parser.py in parse_dot_data(s)
    547         graphparser.parseWithTabs()
--> 548         tokens = graphparser.parseString(s)
    549         return list(tokens)

/usr/local/lib/python3.8/site-packages/pyparsing.py in parseString(self, instring, parseAll)
   1954                     exc.__traceback__ = self._trim_traceback(exc.__traceback__)
-> 1955                 raise exc
   1956         else:

/usr/local/lib/python3.8/site-packages/pyparsing.py in parseImpl(self, instring, loc, doActions)
   3002             return loc + self.matchLen, self.returnString
-> 3003         raise ParseException(instring, loc, self.errmsg, self)
   3004 

ParseException: Expected {'graph' | 'digraph'}  (at char 0), (line:1, col:1)

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-62-40364da769d0> in <module>
      3 
      4 print(maze_code)
----> 5 cg = get_callgraph(maze_code)

~/git/class/class-tva/fuzzingbook/notebooks/ControlFlow.ipynb in get_callgraph(code, name)
    706         construct_callgraph(code, name)
    707     return nx.drawing.nx_pydot.read_dot(name + '.dot')
--> 708 
    709 def generate_print_maze(maze_string):
    710     return """

<decorator-gen-862> in read_dot(path)

/usr/local/lib/python3.8/site-packages/networkx/utils/decorators.py in _open_file(func_to_be_decorated, *args, **kwargs)
    237         # Finally, we call the original function, making sure to close the fobj
    238         try:
--> 239             result = func_to_be_decorated(*new_args, **kwargs)
    240         finally:
    241             if close_fobj:

/usr/local/lib/python3.8/site-packages/networkx/drawing/nx_pydot.py in read_dot(path)
     67 
     68     # List of one or more "pydot.Dot" instances deserialized from this file.
---> 69     P_list = pydot.graph_from_dot_data(data)
     70 
     71     # Convert only the first such instance into a NetworkX graph.

/usr/local/lib/python3.8/site-packages/pydot.py in graph_from_dot_data(s)
    280     @rtype: `list` of `pydot.Dot`
    281     """
--> 282     return dot_parser.parse_dot_data(s)
    283 
    284 

/usr/local/lib/python3.8/site-packages/dot_parser.py in parse_dot_data(s)
    550     except ParseException as err:
    551         print(
--> 552             err.line +
    553             " "*(err.column-1) + "^" +
    554             err)

TypeError: can only concatenate str (not "ParseException") to str
```

**Comment by mboehme** [comment]
Sorry to hear about your experience. I've been talking to @vrthra on another thread, and he mentioned it might be related to Pyan when you run it multiple times and you get some interrupted runs. Can you restart the kernel and see whether this fixes it (temporarily)?

**Comment by darkrsw** [comment]
@mboehme This seems relevant to this pr: https://github.com/uds-se/fuzzingbook/pull/84
In some systems, it is necessary to explicitly install pyan3 and edit the corresponding code.

**Comment by mboehme** [comment]
Thanks @darkrsw!

@nguyenthanhvuh PR #84 added the following line to the greybox fuzzer chapter:
```
Troubleshooting: if nothing is shown after executing `callgraph()`, 
check `pyan` or `pyan3` is correctly installed. If `pyan3` is installed, 
it is necessary to modify `construct_callgraph()` in 
[ControlFlow.ipynb](ControlFlow.ipynb#Example:-Maze) accordingly.)"
```
BTW: @darkrsw What is the concrete modification needed in `ControlFlow.ipynb#Example` when pyan3 is installed?

**Comment by darkrsw** [comment]
@mboehme 
In the current version, it should be `ControlFlow.html#Call-Graph-Helpers`.
Now you can find the change in https://github.com/uds-se/fuzzingbook/pull/87
Please merge the changes. 

Thanks!

**Comment by mboehme** [comment]
Closed via #87.
